pkg/storage/s3.go

```diff
@@ -72,8 +80,28 @@ func (s *S3Storage) objectKey(t tile.TileCoord, prefixOverride string) (string,
 	return interpol.WithMap(s.keyPattern, m)
 }
 
-func (s *S3Storage) respondWithKey(key string, c Condition) (*StorageResponse, error) {
+func (s *S3Storage) respondWithKey(key string, c state.Condition) (*StorageResponse, error) {
 	var result *StorageResponse
+	ctx := context.Background()
+
+	cacheKey := fmt.Sprintf("s3://%s/%s", s.bucket, key)
+	cached, err := s.tileCache.Get(ctx, cacheKey)
```

Comment:
I thought about that, but did it this way because it might not make sense to cache the file stuff. Also, it was easier to build a cache key here where I have the S3 bucket and key. It makes sense to do it up at the handler level, though. I can do that.

Reply:
Fixed in 3ca5a47.